examples/iris_testbed/tests/hdlc_txvr/python_hdlc/hdlc.py

```diff
@@ -262,15 +326,18 @@ def hdlc_rx(self, ser):
 					# TODO; need to pass to the thread register to this port
 					# check the entry (type uart pkt hdr):
 					# notify the app thread data is ready to be received					 
-						print("hdlc: received packet for port %d\n" % data) 
 						self.recv_seq_no += 1
 						self.mutex.release()
-						src_port, dst_port, pkt_type, msgstr = unpack('HHN59s', data)
+						# debug # print ("len of byte obj %d" % len(data))
+						# need to find valid subscriber
+						src_port, dst_port, pkt_type, msgstrenc = unpack('HHB59s', data)
```

Comment:
Could you comment on the byte packing structure you're using? I just want to make sure that the messages you send over the serial devices are in fact bytes. In C, what I did was pack the structs so that there are no padded 0's (C will align structs to the next nearest address space and pad 0s to fill space if you don't ask the compiler to pack the struct, you can google this).

Reply:
Hi Jason,
The reason for that is a header is attached to the message packet we want to transmit in a HDLC frame. The code here is similar to uart_pkt_parse_hdr. The reason I use pack / unpack is because I was afraid if I use a class for the header structure, the size may not be calculated easily

Reply:
Does the code ensure that anything pushed over the emulated serial devices is in bytes and not encoded (e.g. unicode)?

Reply:
Do you mean if the string will be encoded to unicode automatically when pack is called? I don't see that in Python's documentation, so I don't think it will. Do you want to encode string to unicode first then pass it to pack function?